qiskit/providers/aer/backends/backend_utils.py

```diff
@@ -207,3 +207,14 @@ def func(data):
         return {key: func(val) for key, val in data.items()}
 
     return func(data)
+
+
+def circuit_optypes(circuit):
+    """Return set of all operation names in a circuit"""
```

Comment:
They're types, not names - that's rather more specific. It might be good to mention in the docstring that it includes all subclasses as well, so we don't forget / other developers know that the function is safe with regards to subclasses.